opslevel/resource_opslevel_team.go

```diff
@@ -93,6 +101,66 @@ func reconcileTeamAliases(d *schema.ResourceData, team *opslevel.Team, client *o
 	return nil
 }
 
+func collectMembersFromTeam(team *opslevel.Team) []string {
+	members := []string{}
+
+	for _, user := range team.Members.Nodes {
+		members = append(members, user.Email)
+	}
+	return members
+}
+
+func reconcileTeamMembership(d *schema.ResourceData, team *opslevel.Team, client *opslevel.Client) error {
+	expectedMembers := expandStringArray(d.Get("members").(*schema.Set).List())
+	existingMembers := collectMembersFromTeam(team)
+
+	membersToRemove := []string{}
+	membersToAdd := []string{}
+
+	for _, existingMember := range existingMembers {
+		if stringInArray(existingMember, expectedMembers) {
+			continue
+		}
+
+		membersToRemove = append(membersToRemove, existingMember)
+	}
+
+	for _, expectedMember := range expectedMembers {
+
+		if stringInArray(expectedMember, existingMembers) {
+			continue
+		}
+		membersToAdd = append(membersToAdd, expectedMember)
+	}
+
+	if len(membersToAdd) != 0 {
+		_, err := client.AddMembers(&team.TeamId, membersToAdd)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(membersToRemove) != 0 {
+		_, err := client.RemoveMembers(&team.TeamId, membersToRemove)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func validateMembershipState(d *schema.ResourceData) error {
+	if membersSet, ok := d.GetOk("members"); ok {
+		if managerEmail, ok := d.GetOk("manager_email"); ok {
+			memberEmails := expandStringArray(membersSet.(*schema.Set).List())
+			if !stringInArray(managerEmail.(string), memberEmails) {
+				return errors.New("The `manager_email` value is required as a member")
```

Comment:
I really like this! I think there's a few places that I could add this to enforce certain configs.